src/App/Datums.cpp

```diff
@@ -401,3 +387,10 @@ bool LocalCoordinateSystem::LCSExtension::extensionGetSubObject(DocumentObject*&
         return false;
     }
 }
+
+bool LocalCoordinateSystem::hasObject(const DocumentObject* obj, [[maybe_unused]] bool recursive) const
+{
+    auto* lcs = dynamic_cast<const LocalCoordinateSystem*>(getExtendedObject());
```

Comment:
`isDerivedFrom`

Reply:
If the cast can fail, calling upon a nullptr will cause segfault, and nullptr check is needed. If the cast can never fail, it should be static_cast

Reply:
So what you mean here is that I can do a static_cast directly without testing isDerivedFrom because we know the extendedobject is LCS right?

Reply:
There are two problems here:
1. Use of `dynamic_cast` which can be slower than `isDerivedFrom` + `static_cast` - something that @benj5378 brought up recently and we have not considered that earlier. If you want `dynamic_cast` behavior you should use `Base::freecad_dynamic_cast` (soon `freecad_cast` probably) instead which is faster (does `isDerivedFrom` check and then `static_cast` internally).
2. The result of `dynamic_cast` is not checked. If at this point we have guarantee that the object is derived from LocalCoordinateSystem and there is no need to check it you should use `static_cast`. If it is not - the result should be checked. Current code implies that there is no guarantee but it is not checked anyway.

Reply:
Actually we don't even need this variable here.